mscore/scorecmp/scorecmp.cpp

```diff
@@ -106,6 +106,19 @@ void ScoreComparisonTool::showEvent(QShowEvent* e)
 
       updateDiff();
       }
+      
+//---------------------------------------------------------
+//   ScoreComparisonTool::hideEvent
+//---------------------------------------------------------
+
+void ScoreComparisonTool::hideEvent(QHideEvent* e)
+      {
+      QDockWidget::hideEvent(e);
+      if (e->spontaneous())
```

Comment:
What is the purpose of this check?

Reply:
I'm not sure; I just copied the beginning of showEvent(). Should I get rid of it?

Reply:
It prevents unnecessary operations if this widgets gets hidden because of system events, e.g. because the entire window gets minimized, see the documentation [here](https://doc.qt.io/qt-5/qhideevent.html). In this case we don't really care about such show/hide events as we are interested here in whether this widgets gets shown or hidden within the window rather than whether user can actually see this widget at the moment. So it looks like this check here is necessary here: without this check this code will return the previous split screen state even if user just minimizes and shows the window again, which doesn't seem to be intended here.